server/tso/tso.go

```diff
@@ -203,46 +206,51 @@ func (t *timestampOracle) isInitialized() bool {
 }
 
 // resetUserTimestamp update the TSO in memory with specified TSO by an atomically way.
+// When ignoreSmaller is true, resetUserTimestamp will ignore the smaller tso resetting error and do nothing.
+// It's used to write MaxTS during the Global TSO synchronization whitout failing the writing as much as possible.
 func (t *timestampOracle) resetUserTimestamp(leadership *election.Leadership, tso uint64, ignoreSmaller bool) error {
 	t.tsoMux.Lock()
 	defer t.tsoMux.Unlock()
 	if !leadership.Check() {
 		tsoCounter.WithLabelValues("err_lease_reset_ts", t.dcLocation).Inc()
 		return errs.ErrResetUserTimestamp.FastGenByArgs("lease expired")
 	}
-	nextPhysical, nextLogical := tsoutil.ParseTS(tso)
-	var err error
-	// do not update if next logical time is less/before than prev
-	if typeutil.SubTimeByWallClock(nextPhysical, t.tsoMux.physical) == 0 && int64(nextLogical) <= t.tsoMux.logical {
-		tsoCounter.WithLabelValues("err_reset_small_counter", t.dcLocation).Inc()
-		if !ignoreSmaller {
-			err = errs.ErrResetUserTimestamp.FastGenByArgs("the specified counter is smaller than now")
-		}
-	}
+	var (
+		nextPhysical, nextLogical = tsoutil.ParseTS(tso)
+		logicalDifference         = int64(nextLogical) - t.tsoMux.logical
+		physicalDifference        = typeutil.SubTimeByWallClock(nextPhysical, t.tsoMux.physical)
```

Comment:
How about make the accuracy of  `t.tsoMux.physical` always at the millisecond level?

Reply:
If we use `int64` as the `t.tsoMux.physical` here will cause too many code changes because `time.Duration` and `time.Time` are massively used in the TSO code. Using `time.Time` as `t.tsoMux.physical` is more reasonable. However, the previous code is a bit redundant. I change a more clear way. PTAL again!

Reply:
I have a consider about the new change, If I want to change `updateTimestampGuard` to `2ms`, I cannot just adjust the value.

Reply:
I think `updateTimestampGuard` is a different term from the TSO's minimal unit. `updateTimestampGuard` should only be used when we compare the real-world time, such as making the etcd persistence determination. The TSO physical time comparison should always be kept in the unit of 1 millisecond, and I haven't thought of any reason to make the unit bigger yet.

Reply:
There is a scenario that maybe we can consider reducing the accuracy of the physical time when the logical bits are not enough. Now we update `physicalTS`  with `50 ms` interval.  But use millisecond as minimal unit is ok to me.